crates/server/src/verifywalletsignatures/near.rs

```diff
@@ -7,8 +7,51 @@ use base64::engine::Engine;
 use borsh::BorshSerialize;
 use ed25519_dalek::{Signature, Verifier, VerifyingKey};
 use eyre::{eyre, Report, Result as EyreResult};
+use reqwest::{Client, StatusCode};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
 use sha2::{Digest, Sha256};
 
+use crate::admin::service::ApiError;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NearJsonRpcResponse<T> {
+    jsonrpc: String,
+    result: Option<T>,
+    error: Option<NearJsonRpcError>, // Top-level error
+    id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NearJsonRpcError {
+    code: i64,
+    message: String,
+    data: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ResultDataWithPermission {
+    block_hash: String,
+    block_height: u64,
+    nonce: Option<u64>,
+    permission: Option<Permission>,
+    error: Option<String>, // Result-level error
+}
```

Comment:
I think we should be adding Rustdocs to everything new, and this is a good example of where we would definitely benefit from the documentation! 😄